### PR TITLE
release/2.0.0 - Serialization test fixes

### PR DESCRIPTION
**Pull request by ViktorHofer**
Re-enables disabled serialization tests for CompareInfo, CookieContainer and language agnostics types.

cc @danmosemsft 

waiting for shiproom approval

### CODE REVIEW

**Comment by danmosemsft** [comment]
waiting for @weshaggard to declare this branch open for test changes...

**Comment by danmosemsft** [comment]
Diff causing error:
>>m_fqdnMyDomainm_domainTable
<<m_domainTable

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs [hunk 1]
```diff
@@ -407,7 +427,7 @@ public static IEnumerable<object[]> SerializableObjects_MemberData()
 
             // CultureInfo.InvariantCulture.CompareInfo has fields which get lazy initialized
             var sortVersion = new SortVersion(12345, Guid.Parse("9fac6fbb-f683-42f5-a593-0d4a5e34b257"));
-            var compareInfo = new CultureInfo("").CompareInfo;
+            CompareInfo compareInfo = new CultureInfo("").CompareInfo;
```
Comment (tarekgh) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: could use CultureInfo.InvarinatCulture.CompareInfo

File: src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs [hunk 2]
```diff
@@ -20,6 +20,23 @@ public partial class BinaryFormatterTests
...
         private static DateTime s_fixedTimestamp = DateTime.Parse("11/3/1989 04:50:29 AM", CultureInfo.InvariantCulture.DateTimeFormat);
 
+        public static IEnumerable<object[]> SerializableObjects_MemberData()
+        {
+            // Save old culture and set a fixed culture for object instantiation
+            CultureInfo oldCulture = CultureInfo.CurrentCulture;
+
+            try
+            {
+                CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
```
Comment (tarekgh) [inline src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: Is there any way to refactor and run this code in RemoteInvoke? setting the current culture like that can potentially cause some problems for other tests.

Comment (tarekgh) [reply src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTestData.cs]: I know you are porting the changes. I just talking in general refactoring the tests on master

**Comment by tarekgh** [comment]
LGTM

**Comment by ViktorHofer** [comment]
Thanks Tarek for the review, I will take it into consideration into master.

I'm closing this PR because I don't see any value in porting it into the Core 2.0 branch. It only reenables the test for CompareInfo which isn't necessary at this point. It doesn't fix any test errors (like the one in the UWP6.0 branch, but we also closed it as we aren't concerned about non UWP errors there).

**Comment by ViktorHofer** [comment]
> My guess is the binary serializer has a dictionary for these, and it's using a different comparer?

No, the order is just swapped in the second one. This happens only in the release/2.0.0 branch when you use reflection to change some fields. I already spent too much time about a month ago to investigate that behavior and I didn't find the root root cause for it.

**Pull request closed without merging**